src/toil/batchSystems/abstractBatchSystem.py

```diff
@@ -293,8 +293,12 @@ def workerCleanup(info):
         """
         assert isinstance(info, WorkerCleanupInfo)
         workflowDir = Toil.getWorkflowDir(info.workflowID, info.workDir)
-        if (info.cleanWorkDir == 'always'
-            or info.cleanWorkDir in ('onSuccess', 'onError') and os.listdir(workflowDir) == []):
+        workflowDirContents = os.listdir(workflowDir)
+        from toil.job import cacheDirName
+        if (info.cleanWorkDir == 'always' or
+                info.cleanWorkDir in ('onSuccess', 'onError') and
+                    (workflowDirContents == [] or
```

Comment:
I think you can replace this disjunction with a use of the `in` operator. Comparing against list literals isn't the most efficient method but it is very concise.